### PR TITLE
Harvester arm driver

### PR DESCRIPTION
**Pull request by ibrokethecloud**
## Issue: 

Related Harvester issue: https://github.com/harvester/harvester/issues/6053
Related Rancher issue: https://github.com/rancher/rancher/issues/47238

## Problem

  Currently rancher builds package amd64 only driver in the multi-arch builds. As a result end users running Rancher on arm64 hosts cannot provision downstream clusters on harvester since the driver is hardcoded to amd64.
## Solution

Harvester node driver v0.6.9 now publishes multi-arch driver images and the change introduced packages the correct driver arch in rancher image builds.
In addition, the additional check in machinedriver_data.go ensures the arch appropriate checksum is used.

### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

## QA Testing Considerations

 To test:
* Install rancher on arm64 vm
* Import a harvester cluster
* provision a downstream cluster on harvester, and this should be successful

Repeat same test by installing rancher on an amd64 vm

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by innobead — DISMISSED** [review]
In general, LGTM.

However, the current implementation of MachineDriver mgmt seems not consider multi-arch design, so the PR here is more just fix specially for Harvester. Not sure how other drivers will be handled in the future, as notice several are hard-coded with amd64 artifact. 

P.S. For checksums, it's better to publish them to the release page of the Harvester node driver as well, even though this is out of scope here.

**Comment by innobead** [comment]
cc @khushboo-rancher

File: pkg/data/management/machinedriver_data.go [hunk 1]
```diff
@@ -5,14 +5,16 @@ import (
 	"os"
 	"os/exec"
 	"reflect"
+	goruntime "runtime"
```
Comment (jakefhyde) [inline pkg/data/management/machinedriver_data.go:8]:
```suggestion
	"runtime"
```
Nit: runtime is self-explanatory IMO

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Minor nit, otherwise LGTM. As others have said, this is probably fine for harvester in the meantime, extending this to other drivers will likely require something more involved (feature enhancement or something).

**Comment by ibrokethecloud** [comment]
may be we should move to inject driver file checksums as build variables, this can then be done in docker build based on arch being built and file being downloaded

**Review by jakefhyde — DISMISSED** [review]

**Review by bk201 — APPROVED** [review]

File: pkg/data/management/machinedriver_data.go [hunk 2]
```diff
@@ -5,14 +5,16 @@ import (
 	"os"
 	"os/exec"
 	"reflect"
+	"runtime"
 	"strings"
 
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
```
Comment (Oats87) [inline pkg/data/management/machinedriver_data.go:13]: unnecessary newline

**Review by Oats87 — CHANGES_REQUESTED** [review]

**Comment by m-ildefons** [comment]
Just my two cents:

There has been [this PR](https://github.com/harvester/docker-machine-driver-harvester/pull/61) since the last release, which provides a fix that is quite critical to get [this workaround](https://harvesterhci.io/kb/renew_harvester_cloud_credentials/) to work right.
We should do another release of the Harvester node driver, which includes the fix. That would mean, that we would need to update this PR to include that.

It may be out of scope for now, but I think it would be worthwhile to consider [injecting the version and checksums as build time variables](https://blog.alexellis.io/inject-build-time-vars-golang/) instead of hard-coding them in the Golang code.
Then the Dockerfile would serve as the single source of truth and updates would become much easier, since no changes to the Golang code would be necessary.

**Review by Oats87 — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**